content/sensu-go/5.0/reference/rbac.md

```diff
@@ -1,132 +1,298 @@
+{{</highlight>}}
+
+_WARNING: This deletes every resource definition associated with the namespace._
+
+To get help managing namespaces with sensuctl:
+
+{{< highlight shell >}}
+sensuctl namespace help
+{{</highlight>}}
+
+### Namespace specification
 
 #### Attributes
 
 name         | 
 -------------|------ 
-description  | The name of the environment. Cannot contain special characters.
+description  | The name of the namespace. Names can contain alphanumeric characters and hyphens, but must begin and end with an alphanumeric character.
 required     | true 
 type         | String
-example      | {{< highlight shell >}}"name": "default"{{</highlight>}}
+example      | {{< highlight shell >}}"name": "ops"{{</highlight>}}
 
 description  | 
 -------------|------ 
-description  | Description of the environment 
+description  | Description of the namespace
 required     | false 
 type         | String
-example      | {{< highlight shell >}}"description": "Default environment"{{</highlight>}}
+example      | {{< highlight shell >}}"description": "Ops team production monitoring"{{</highlight>}}
 
-organization | 
--------------|------ 
-description  | The name of the organization the environment belongs to. 
-required     | false 
-type         | String
-default      | current environment value configured for `sensuctl` (ie `default`)
-example      | {{< highlight shell >}}"organization": "default"{{</highlight>}}
+## Resources
+Permissions within Sensu are scoped to resource types, like checks, handlers, and users.
+All resource types (with the exceptions of namespaces, users, and roles) must belong to a single namespace.
 
-### User
+### Resource types
 
-A user represents a person or an agent which interacts with Sensu.
+| Type | Description |
+|---|---|
+| `*` | All resources within Sensu. **The `*` type takes precedence over other rules within the same role.** If you wish to deny a certain type, you can't use the `*` type and must explicitly allow every type required |
+| `assets` | [Asset][5] resources within a namespace |
+| `checks` | [Check][6] resources within a namespace |
+| `entities` | [Entity][7] resources within a namespace |
+| `events` | [Event][8] resources within a namespace |
+| `handlers` | [Handler][9] resources within a namespace |
+| `hooks` | [Hook][10] resources within a namespace |
+| `mutators` | [Mutator][11] resources within a namespace |
+| `namespace` | [Namespaces][12] |
+| `roles` | [Roles][13] |
+| `silenced` | [Silencing][14] resources within a namespace |
+| `users` | [Users][15] |
+
+### Assigning a resource to a namespace
+
+You can assign a resource to a namespace in the resource definition.
+Namespaces, users, and roles are part of Sensu RBAC and don't need to be assigned to a namespace.
+
+For example, to assign a check called `check-cpu` to the `ops` namespace, include the `namespace` attribute in the check definition:
+
+{{< highlight json >}}
+{
+  "type": "CheckConfig",
```

Comment:
Will soon be replaced by `kind`, see https://github.com/sensu/sensu-go/issues/2320